### DISCUSSION TITLE
version agnostic pep582 script path?

### DISCUSSION THREAD

**Discussion by songololo** [pdm-project/pdm#1067] [Ideas]
Is it possible to create a workflow for using a version agnostic pep582 initialisation path?

I'm referring to the `pep582` script which gets injected to `.zshrc` (Mac)

When upgrading PDM via `brew` the new version number invalidates the existing script path, e.g. `1.14.1` would render this path useless when `pdm` is next updated.

```bash
export PYTHONPATH='/usr/local/Cellar/pdm/1.14.1/libexec/lib/python3.10/site-packages/pdm/pep582`
``` 

Is there a way to create a version agnostic path so that we don't need to manually update the configs?

EDIT - I guess the same is true for the `.pdm.toml` file, which means that each time brew updates `pdm` we need to check that `pdm.toml` files are pointing to the latest version?

```toml
[python]
path = "/usr/local/Cellar/pdm/1.14.1/libexec/bin/python3.10"
```

**Comment by frostming** [comment]
It is homebrew's fault to put a version in the path, isn't it?

For PDM installed by the install script, it should be `$HOME/.local/pdm/venv/lib/python/X.Y/site-packages/pdm/pep582`

The only varying part is the python version `X.Y`, which won't be updated that frequently.

**Reply by songololo** [reply]
Just thinking, for `brew`, wouldn't the `opt` path be an option? If I understand correctly, brew will link the latest version via `opt` path symlinks?

i.e. wouldn't it be OK to use this instead?

```bash
/usr/local/opt/pdm/libexec/lib/python3.10/site-packages/pdm/pep582
```
